app/models/dataset_record/darwin_core/occurrence.rb

```diff
@@ -211,6 +211,17 @@ def import(dwc_data_attributes = {})
 
         # TODO: If all attributes are equal assume it is the same event and share it with other specimens? (eventID is an alternate method to detect duplicates)
         if collecting_event
+          # if tags have been specified to be added, update the collecting event
+          if attributes[:collecting_event][:tags_attributes]
+            # get list of preexisting tags, exclude them from update
+            current_tags = collecting_event.tags.pluck(:keyword_id).to_set
+
+            new_tags = attributes[:collecting_event][:tags_attributes].reject { |t| current_tags.member?(t[:keyword].id)}
+
+            # add tags if there were any new ones
+            collecting_event.update!(tags_attributes: new_tags) if new_tags
```

Comment:
I think this will destroy existing tags too. Is this intended?

Reply:
This will not destroy existing tags, unless `_destroy: true` IIRC